cfme/tests/networks/test_sdn_inventory_collection.py

```diff
@@ -130,14 +131,16 @@ def test_sdn_nsg_firewall_rules(provider, appliance, secgroup_with_rule):
     # Navigate to network provider.
     collection = appliance.collections.network_providers.filter({'provider': provider})
     network_provider = collection.all()[0]
+    network_provider.refresh_provider_relationships()
+    wait_for(network_provider.is_refreshed, func_kwargs=dict(refresh_delta=10), timeout=600)
     view = navigate_to(network_provider, 'Details')
     parent_name = view.entities.relationships.get_text_of("Parent Cloud Provider")
     assert parent_name == provider.name
 
-    # Navigate to secgroup.
-    collection = appliance.collections.network_security_groups.filter({'name': secgroup_with_rule})
+    collection = appliance.collections.network_security_groups.filter(
+        {'name': secgroup_with_rule})
     secgroup = collection.all()[0]
     view = navigate_to(secgroup, 'Details')
 
-    assert 'Port Range' == view.entities.firewall_rules[0][3].text
-    assert '22' == view.entities.firewall_rules[1][3].text
+    assert 'Port' == view.entities.firewall_rules[1][3].text
+    assert '22' == view.entities.firewall_rules[2][3].text
```

Comment:
Required:
assert int(view.entities.firewall_rules[2][3].text) == 22

Reply:
Your suggestion does exactly the same as my version, but it is a bit more complex .

Reply:
I think; Numerical thing should not compare as a string.

Reply:
@digitronik `'22'` is string. `'22' == '22'` is perfectly ok. Doing `int('22') == 22` is just adding additional step

Reply:
:P, @jaryn its simple to understand and the suggestion is for good practice. `firewall rule port` is number like port `8080`. When we assert such things we should have to convert `string` to mathematical quantity before assert. The thumb rule is that always do operations in the same domain what it represents. 
`"22"` represents a specific ASCII symbol (also known as a "character")
`22` represent a mathematical number.

In this case its ok but if we read `"22"` l.e. `string`. It's most likely the mathematical quantity, but it could have different meanings depending on what other characters are nearby. like "jaryn22" :). You are free to choose what you like...